cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -306,18 +308,61 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
     poolsDir = outputDir </> "pools-keys"
     stakeDelegatorsDir = outputDir </> "stake-delegators"
-    numStakeDelegators = case stakeDelegators of OnDisk n -> n; Transient n -> n
     mkDelegationMapEntry :: Delegation -> (L.KeyHash L.Staking L.StandardCrypto, L.PoolParams L.StandardCrypto)
     mkDelegationMapEntry d = (dDelegStaking d, dPoolParams d)
 
+    addDRepsToConwayGenesis :: [VerificationKey DRepKey] -> [VerificationKey StakeKey]
+                            -> L.ConwayGenesis L.StandardCrypto -> L.ConwayGenesis L.StandardCrypto
+    addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
+      conwayGenesis { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+                    , L.cgInitialDReps = initialDReps dRepKeys
+                    }
+
+      where
+      delegs :: [(VerificationKey StakeKey, VerificationKey DRepKey)] -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      delegs = ListMap.fromList . map (bimap verificationKeytoStakeCredential (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential))
+
+      initialDReps :: [VerificationKey DRepKey] -> ListMap (L.Credential L.DRepRole L.StandardCrypto) (L.DRepState L.StandardCrypto)
+      initialDReps = ListMap.fromList . map (\c -> ( verificationKeyToDRepCredential c
+                                                   , L.DRepState { L.drepExpiry = EpochNo 1000
+                                                                 , L.drepAnchor = SNothing
+                                                                 , L.drepDeposit = L.Coin 100000000
+                                                                 }))
+
+      verificationKeyToDRepCredential :: VerificationKey DRepKey -> L.Credential L.DRepRole L.StandardCrypto
+      verificationKeyToDRepCredential vk = dRepKeyToCredential (verificationKeyHash vk)
+        where
+          dRepKeyToCredential :: Hash DRepKey -> L.Credential L.DRepRole L.StandardCrypto
+          dRepKeyToCredential (DRepKeyHash v) = L.KeyHashObj v
+
+      verificationKeytoStakeCredential :: VerificationKey StakeKey -> L.Credential L.Staking L.StandardCrypto
+      verificationKeytoStakeCredential vk = stakeKeyToCredential (verificationKeyHash vk)
+        where
+          stakeKeyToCredential :: Hash StakeKey -> L.Credential L.Staking L.StandardCrypto
+          stakeKeyToCredential (StakeKeyHash v) = L.KeyHashObj v
+
+    zipWithDeepSeq :: NFData c => (a -> b -> c) -> [a] -> [b] -> [c]
+    zipWithDeepSeq _ _ [] = []
+    zipWithDeepSeq _ [] _ = []
+    zipWithDeepSeq f (h1:t1) (h2:t2) = let h = f h1 h2 in
+                                       h `deepseq` (h:zipWithDeepSeq f t1 t2)
+
+    mapAccumM :: (acc -> b -> IO (acc, c)) -> acc -> [b] -> IO [c]
+    mapAccumM _ _ [] = return []
+    mapAccumM f a (h:t) = do (a', h') <- f a h
+                             rest <- mapAccumM f a' t
+                             return $ h':rest
```

Comment:
Is the [mapAccumM](https://hackage.haskell.org/package/base-4.19.1.0/docs/Data-Traversable.html#v:mapAccumM) implementation from base inadequate? If yes, a comment would help remembering why you wrote your own here.

Reply:
Turns out I measured after replacing it, and it seems it works just as efficiently. I must have done two changes at once and thought it was an improvement because of the other change. I will remove it and use the standard one